File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        
        if(head == null || head.next == null) return head;

        ListNode prev = head;
        ListNode curr = head.next;
        ListNode next = curr.next;

        prev.next = null;

        while(next != null) {
            System.out.println(curr.val);
            curr.next = prev;
            prev = curr;
            curr = next;
            next = next.next;
        }
        curr.next = prev;

        return curr;
    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if len(s) == 0 or len(s) == 1:
            return s
        
        longest = s[0]
        curr = ""
        for i in range(len(s)):
            curr += s[i]
            for j in range(i+1, len(s)):
                curr += s[j]
                if curr == curr[::-1] and len(curr) > len(longest):
                    longest = curr
            curr = ""

        return longest